x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { SettingsPage } from './SettingsPage/SettingsPage';
+import { fromQuery, toQuery } from '../../../../shared/Links/url_helpers';
+
+type PageStep = 'choose-service-step' | 'choose-settings-step' | 'review-step';
+
+function setPage(pageStep: PageStep) {
+  history.push({
+    ...history.location,
+    search: fromQuery({
+      ...toQuery(history.location.search),
+      pageStep
+    })
+  });
+}
+
+function getUnsavedChanges({
+  newConfig,
+  existingConfig
+}: {
+  newConfig: AgentConfigurationIntake;
+  existingConfig?: AgentConfigurationIntake;
+}) {
+  return Object.fromEntries(
+    Object.entries(newConfig.settings).filter(([key, value]) => {
+      const existingValue = existingConfig?.settings?.[key];
+
+      // don't highlight changes that were added and removed
+      if (value === '' && existingValue == null) {
+        return false;
+      }
+
+      return existingValue !== value;
+    })
+  );
+}
+
+export function AgentConfigurationCreateEdit({
+  pageStep,
+  existingConfig
+}: {
+  pageStep: PageStep;
+  existingConfig?: AgentConfigurationIntake;
+}) {
+  const [newConfig, setNewConfig] = useState<AgentConfigurationIntake>({
+    agent_name: existingConfig?.agent_name,
+    service: existingConfig?.service || {},
+    settings: existingConfig?.settings || {}
+  });
+  const isEditMode = Boolean(existingConfig);
+
+  useEffect(() => {
+    // the user tried to edit the service of an existing config
+    if (pageStep === 'choose-service-step' && isEditMode) {
+      setPage('choose-settings-step');
+    }
+  }, [isEditMode, newConfig, pageStep]);
+
+  const unsavedChanges = getUnsavedChanges({ newConfig, existingConfig });
+
+  return (
+    <>
+      <EuiTitle>
+        <h2>
+          {isEditMode
+            ? i18n.translate('xpack.apm.agentConfig.editConfigTitle', {
+                defaultMessage: 'Edit configuration'
+              })
+            : i18n.translate('xpack.apm.agentConfig.createConfigTitle', {
+                defaultMessage: 'Create configuration'
+              })}
+        </h2>
+      </EuiTitle>
+
+      <EuiText size="s">
+        This allows you to fine-tune your agent configuration directly in
+        Kibana. Best of all, changes are automatically propagated to your APM
+        agents so there’s no need to redeploy.
+      </EuiText>
+
+      <EuiSpacer size="m" />
+
+      {pageStep === 'choose-service-step' && (
+        <ServicePage
+          newConfig={newConfig}
+          setNewConfig={setNewConfig}
+          onClickNext={() => setPage('choose-settings-step')}
+        />
+      )}
+
+      {pageStep === 'choose-settings-step' && (
+        <SettingsPage
+          unsavedChanges={unsavedChanges}
+          onClickEdit={() => setPage('choose-service-step')}
+          newConfig={newConfig}
+          setNewConfig={setNewConfig}
+          isEditMode={isEditMode}
+        />
+      )}
+
+      {pageStep === 'review-step' && <div>Review will be here </div>}
```

Comment:
Should this be marked more clearly as TODO so we don't forget about it?

Reply:
Good point. I'll comment it out for now